otter/test/log/test_intents.py

```diff
@@ -84,6 +84,25 @@ def test_err(self):
         self.assertIsNone(r)
         self.log.err.assert_called_once_with(f, "yo!", f1='v')
 
+    def test_err_from_context(self):
+        """
+        When None is passed as the failure, the exception comes from the
+        context at the time of creating the intent, not the time at which the
+        intent is performed.
+        """
+        try:
+            raise RuntimeError('original')
+        except RuntimeError:
+            eff = err(None, "why")
+        try:
+            raise RuntimeError('performing')
+        except RuntimeError:
+            sync_perform(self.disp, eff)
+        self.log.err.assert_called_once_with(
+            transform_eq(lambda f: (f.type, str(f.value)),
```

Comment:
Non-blocking thought for FYI only: You could use `CheckFailureValue` in this case but even better would be to implement that using `transform_eq` :)